src/main/webapp/js/map-loader.js

```diff
@@ -1,38 +1,133 @@
+let editMarker;
+let map;
+let displayMarkers = {};
  /**
  * Adds a new marker with a toggable info window
- * @param {Map} map 
  * @param {Number} lat 
  * @param {Number} lng 
- * @param {String} title 
  * @param {String} description 
  */
-function addLandmark(map, lat, lng, title, description){
+function addDisplayMarker(lat, lng, description){
   const marker = new google.maps.Marker({
     position: { lat: lat, lng: lng },
-    map: map,
-    title: title
+    map: map
   })
   const infoWindow = new google.maps.InfoWindow({
-    content: description
+    content: buildDeletableMarker( lat, lng, description)
   })
   marker.addListener('click', () => {
     infoWindow.open(map, marker)
   })
+
+  // Uses a dictionary with both lat and lng for faster lookup
+  if(!displayMarkers[lat]) displayMarkers[lat] = {};
+  displayMarkers[lat][lng] = marker;
 }
 
 /**
  * Main function
  */
 function createMap(){
-  const map = new google.maps.Map(document.getElementById('map'), {
+  map = new google.maps.Map(document.getElementById('map'), {
     center: {lat: 37.764419, lng: -122.445911},
     zoom: 12
   });
+  map.addListener('click', (event) => {
+    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+  })
 
-  addLandmark(map, 37.819021, -122.477946, "Golden Gate Bridge", "This is the Golden Gate Bridge")
-  addLandmark(map, 37.802122, -122.418805, "Lombard Street", "This is Lombard St.")
-  addLandmark(map, 37.811039, -122.415175, "Fishermans Wharf", "This is Fishermans Wharf")
+  fetchMarkers();
+}
 
+ /** Fetches markers from the backend and adds them to the map. */
+function fetchMarkers(){
+  fetch('/markers').then((response) => {
+    return response.json();
```

Comment:
Nit: You can just do (response) => response.json() here

Reply:
Fixed